File: Julia/gan-generator.jl
function generator(z, output_dim::Int)
    noise_dim = size(z, 2)
    W1 = randn(noise_dim, 128) .* 0.02
    b1 = zeros(128)
    W2 = randn(128, output_dim) .* 0.02
    b2 = zeros(output_dim)

    h1 = max.(0, z * W1 .+ b1)
    tanh.(h1 * W2 .+ b2)
end

File: R/gan-generator.R
generator <- function(z, output_dim) {
  noise_dim <- ncol(z)
  W1 <- matrix(rnorm(noise_dim * 128, sd = 0.02), nrow = noise_dim, ncol = 128)
  b1 <- numeric(128)
  W2 <- matrix(rnorm(128 * output_dim, sd = 0.02), nrow = 128, ncol = output_dim)
  b2 <- numeric(output_dim)

  h1 <- pmax(0, z %*% W1 + b1)
  tanh(h1 %*% W2 + b2)
}
